### PR TITLE
Swipe to Delete (Wearable): add example and implementation

### PR DESCRIPTION
**Pull request by grzegorz-czajkowski**
[Issue] https://github.com/Samsung/TAU/issues/1059
[Problem] No support for swipping list item
[Solution] Add example and implementation

Signed-off-by: Grzegorz Czajkowski <g.czajkowski@samsung.com>

### CODE REVIEW

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less [hunk 1]
```diff
@@ -78,7 +78,38 @@
...
+				width: 98 * @px_base;
+				height: 100%;
+				right: 0;
+
+				&-background {
+					position: absolute;
+					width: 98 * @px_base;
+					height: 100%;
+					background: #FFFFFF;
+					border-top-right-radius: 54px;
```
Comment (DohyungLim) [inline src/css/profile/wearable/changeable/theme-circle/arclistview.less]: need change px to @px_base;

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less [hunk 2]
```diff
@@ -78,7 +78,38 @@
...
+					border-bottom-right-radius: 54px;
```
Comment (DohyungLim) [inline src/css/profile/wearable/changeable/theme-circle/arclistview.less]: need change px to @px_base;

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less [hunk 3]
```diff
@@ -78,7 +78,38 @@
 				margin-top: 0;
 			}
 		}
-		
+
+		&.li-swipe-left {
+			.ui-swipe-icon {
+				position: absolute;
+				width: 98 * @px_base;
```
Comment (DohyungLim) [inline src/css/profile/wearable/changeable/theme-circle/arclistview.less:85]: just question.
why use width value 98? and &-background same using width value?

**Review by DohyungLim — COMMENTED** [review]
3 comment.

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less [hunk 4]
```diff
@@ -78,7 +78,38 @@
 				margin-top: 0;
 			}
 		}
-		
+
+		&.li-swipe-left {
```
Comment (TomaszLukawskiSam) [inline src/css/profile/wearable/changeable/theme-circle/arclistview.less]: Add prefix `ui-` for TAU selectors

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less (hunk 1 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/wearable/changeable/theme-circle/arclistview.less]: Fixed.

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less (hunk 2 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/wearable/changeable/theme-circle/arclistview.less]: Fixed.

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less (hunk 3 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/wearable/changeable/theme-circle/arclistview.less:85]: I changed this to have 100% of parent.

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less (hunk 4 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/wearable/changeable/theme-circle/arclistview.less]: Added.

**Comment by grzegorz-czajkowski** [comment]
This PR shows delete icon (using clip-path property) if element is dragged. Many issues has been raveled for which separate issues were created: https://github.com/Samsung/TAU/issues/1227, https://github.com/Samsung/TAU/issues/1228, https://github.com/Samsung/TAU/issues/1229, https://github.com/Samsung/TAU/issues/1230, https://github.com/Samsung/TAU/issues/1231.

File: examples/wearable/UIComponents/contents/presentation-views/list/main/swipe-to-delete.html [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	<link href="../../../../css/style.css" rel="stylesheet" />
+	<link href="../../../../css/style.circle.css" rel="stylesheet" />
+	<script src="../../../../js/circle-helper.js">
+	</script>
+	<script src="../../../../lib/tau/wearable/js/tau.js">
+	</script>
+</head>
+
+<body>
+	<div class="ui-page">
```
Comment (TomaszLukawskiSam) [inline examples/wearable/UIComponents/contents/presentation-views/list/main/swipe-to-delete.html]: You can add page id eg. `id="demo-swipe-to-delete"`

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less [hunk 6]
```diff
@@ -78,7 +78,39 @@
...
+				// https://css-tricks.com/non-transparent-elements-inside-transparent-elements/
+				&-delete {
+					position: absolute;
+					content: '';
+					width: 76 * @px_base;
+					height: 76 * @px_base;
+					left: 50%;
+					top: 50%;
+					transform: translate(-50%, -50%);
+					background: rgba(255, 0, 255, 0.5) url('images/tw_list_delete_ic.png');
```
Comment (TomaszLukawskiSam) [inline src/css/profile/wearable/changeable/theme-circle/arclistview.less]: In my opinion background color is not part of framework this is only for guideline needs

File: src/js/profile/wearable/widget/wearable/ArcListview.js [hunk 7]
```diff
@@ -1432,9 +1442,54 @@
...
 							}
 						}
+
 					}
 				}
-			};
+			}
+
+			prototype._enableSwipe = function (element) {
+				ns.event.enableGesture(
```
Comment (TomaszLukawskiSam) [inline src/js/profile/wearable/widget/wearable/ArcListview.js:1452]: You enabled `drag` gesture but you never disabled it.

File: src/js/profile/wearable/widget/wearable/ArcListview.js [hunk 8]
```diff
@@ -1432,9 +1442,54 @@
...
+
+			prototype._itemDragging = function (event) {
+				var target = event.target,
+					gesture = event.detail,
+					swipeIcon = target.querySelector(selectors.SWIPE_ICON),
+					currentClipPath,
+					newClipPath;
+
+				if (swipeIcon) {
+					currentClipPath = getComputedStyle(swipeIcon).clipPath.match(/[0-9]+%/);
```
Comment (TomaszLukawskiSam) [inline src/js/profile/wearable/widget/wearable/ArcListview.js]: You can define local variable like `PERCENT_REGEXP = /[0-9]+%/`

**Review by TomaszLukawskiSam — CHANGES_REQUESTED** [review]

**Comment by TomaszLukawskiSam** [comment]
> This PR shows delete icon (using clip-path property) if element is dragged. Many issues has been raveled for which separate issues were created: #1227, #1228, #1229, #1230, #1231.

I think there are more issues eg.: Disable horizontal end effect on drag

**Comment by grzegorz-czajkowski** [comment]
> > This PR shows delete icon (using clip-path property) if element is dragged. Many issues has been raveled for which separate issues were created: #1227, #1228, #1229, #1230, #1231.
> 
> I think there are more issues eg.: Disable horizontal end effect on drag

Created issue https://github.com/Samsung/TAU/issues/1234

File: examples/wearable/UIComponents/contents/presentation-views/list/main/swipe-to-delete.html (hunk 5 above)
Comment (grzegorz-czajkowski) [reply examples/wearable/UIComponents/contents/presentation-views/list/main/swipe-to-delete.html]: Added.

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less (hunk 6 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/wearable/changeable/theme-circle/arclistview.less]: Replaced this ui-swipte-icon-background with ui-swipte-icon::after in less only. Now, example looks like:
`                        <div class="ui-swipe-icon">`
    `                            <div class="ui-swipe-icon-delete"></div>`
                        `</div>`
Thanks.

File: src/js/profile/wearable/widget/wearable/ArcListview.js (hunk 8 above)
Comment (grzegorz-czajkowski) [reply src/js/profile/wearable/widget/wearable/ArcListview.js]: Added.

File: src/js/profile/wearable/widget/wearable/ArcListview.js (hunk 7 above)
Comment (grzegorz-czajkowski) [reply src/js/profile/wearable/widget/wearable/ArcListview.js:1452]: Yes, you're right. Drag events are enabled once element is selected. We should disable them once element is unselected. The problem is that the element is unselected immediately after touch/mouse move before drag is started.

According to guide the selection should not disappear during swipe and there is a separate issue for that https://github.com/Samsung/TAU/issues/1227

File: src/css/profile/wearable/changeable/theme-circle/arclistview.less (hunk 6 above)
Comment (TomaszLukawskiSam) [reply src/css/profile/wearable/changeable/theme-circle/arclistview.less]: I have on mind this pink background color. I think it shouldn't be part of framework.

Comment (grzegorz-czajkowski) [reply src/css/profile/wearable/changeable/theme-circle/arclistview.less]: Ok, removed.

**Review by TomaszLukawskiSam — APPROVED** [review]
Looks good to me.

**Pull request merged**